chain/network/src/codec.rs

```diff
@@ -74,12 +74,63 @@ pub fn bytes_to_peer_message(bytes: &[u8]) -> Result<PeerMessage, std::io::Error
     PeerMessage::try_from_slice(bytes)
 }
 
+fn peer_id_type_field_len(enum_var: u8) -> Option<usize> {
+    // 1 byte for enum variant, then some number depending on the
+    // public key type
+    match enum_var {
+        0 => Some(1 + 32),
+        1 => Some(1 + 64),
+        _ => None,
+    }
+}
+
+pub fn is_forward_tx(bytes: &[u8]) -> bool {
```

Comment:
for this particular function, what prevents malicious actor from constructing bytes according to this function so that it returns `true` but the bytes do not actually deserialize to a transaction?

Reply:
An attacker could do that, but it would not accomplish anything.

The counter tracking the number of transactions is only incremented after successful deserialization, so an attacker could not artificially cause the node to reject transactions early. And if the node were to already be under load then any message which returns `true` from `is_forward_tx` will be dropped so again the attacker is not accomplishing anything.

I think the worst thing an attacker could do would be to send many well-formed transaction messages (so they correctly deserialized), but where they are not valid transactions. Then the node would stop accepting any more transaction messages until the next block, but no new transactions would have been added to the mempool. Maybe we need more logic to ban peers which attempt such attacks?